src/idc/idc.c

```diff
@@ -120,9 +121,18 @@ int idc_wait_in_blocking_mode(uint32_t target_core, bool (*cond)(int))
 static void idc_ipc(void)
 {
 	struct ipc *ipc = ipc_get();
-	ipc_cmd_hdr *hdr = ipc->comp_data;
 
-	ipc_cmd(hdr);
+	/*
+	 * In principle we could just do
+	 * ipc->core = cpu_get_id();
+	 * because currently that's the only field that is modified at run-time,
+	 * but flushing the complete cache is more future-proof.
+	 */
+	dcache_invalidate_region(ipc, sizeof(*ipc));
```

Comment:
Does ipc point to the mailbox or to a local core 0 copy of the mailbox ? the latter will need a wb+inv.

Reply:
@lgirdwood ipc points to a global IPC context and it's allocated as "shared," so, actually no manual cache-management is required for it. Sorry, I actually forgot about this PR and instead fixed this commit in #4471 - let's close this PR and continue there.